core/src/main/java/com/nisovin/magicspells/util/AttributeUtil.java

```diff
@@ -46,6 +46,53 @@ public class AttributeUtil {
 			attributeNameMap.put(rep, attribute);
 			attributeNameMap.put(rep.substring(rep.indexOf('.') + 1), attribute);
 		}
+
+		loadLegacyAttributeNames();
+	}
+
+	private static void loadLegacyAttributeNames() {
+		Map<String, Attribute> legacy = new HashMap<>();
+		legacy.put("generic.max_health", Attribute.GENERIC_MAX_HEALTH);
+		legacy.put("generic.follow_range", Attribute.GENERIC_FOLLOW_RANGE);
+		legacy.put("generic.knockback_resistance", Attribute.GENERIC_KNOCKBACK_RESISTANCE);
+		legacy.put("generic.movement_speed", Attribute.GENERIC_MOVEMENT_SPEED);
+		legacy.put("generic.flying_speed", Attribute.GENERIC_FLYING_SPEED);
+		legacy.put("generic.attack_damage", Attribute.GENERIC_ATTACK_DAMAGE);
+		legacy.put("generic.attack_knockback", Attribute.GENERIC_ATTACK_KNOCKBACK);
+		legacy.put("generic.attack_speed", Attribute.GENERIC_ATTACK_SPEED);
+		legacy.put("generic.armor", Attribute.GENERIC_ARMOR);
+		legacy.put("generic.armor_toughness", Attribute.GENERIC_ARMOR_TOUGHNESS);
+		legacy.put("generic.fall_damage_multiplier", Attribute.GENERIC_FALL_DAMAGE_MULTIPLIER);
+		legacy.put("generic.luck", Attribute.GENERIC_LUCK);
+		legacy.put("generic.max_absorption", Attribute.GENERIC_MAX_ABSORPTION);
+		legacy.put("generic.safe_fall_distance", Attribute.GENERIC_SAFE_FALL_DISTANCE);
+		legacy.put("generic.scale", Attribute.GENERIC_SCALE);
+		legacy.put("generic.step_height", Attribute.GENERIC_STEP_HEIGHT);
+		legacy.put("generic.gravity", Attribute.GENERIC_GRAVITY);
+		legacy.put("generic.jump_strength", Attribute.GENERIC_JUMP_STRENGTH);
+		legacy.put("generic.burning_time", Attribute.GENERIC_BURNING_TIME);
+		legacy.put("generic.explosion_knockback_resistance", Attribute.GENERIC_EXPLOSION_KNOCKBACK_RESISTANCE);
+		legacy.put("generic.movement_efficiency", Attribute.GENERIC_MOVEMENT_EFFICIENCY);
+		legacy.put("generic.oxygen_bonus", Attribute.GENERIC_OXYGEN_BONUS);
+		legacy.put("generic.water_movement_efficiency", Attribute.GENERIC_WATER_MOVEMENT_EFFICIENCY);
+		legacy.put("player.block_interaction_range", Attribute.PLAYER_BLOCK_INTERACTION_RANGE);
+		legacy.put("player.entity_interaction_range", Attribute.PLAYER_ENTITY_INTERACTION_RANGE);
+		legacy.put("player.block_break_speed", Attribute.PLAYER_BLOCK_BREAK_SPEED);
+		legacy.put("player.mining_efficiency", Attribute.PLAYER_MINING_EFFICIENCY);
+		legacy.put("player.sneaking_speed", Attribute.PLAYER_SNEAKING_SPEED);
+		legacy.put("player.submerged_mining_speed", Attribute.PLAYER_SUBMERGED_MINING_SPEED);
+		legacy.put("player.sweeping_damage_ratio", Attribute.PLAYER_SWEEPING_DAMAGE_RATIO);
+		legacy.put("zombie.spawn_reinforcements", Attribute.ZOMBIE_SPAWN_REINFORCEMENTS);
+
+		attributeNameMap.putAll(legacy);
+
+		for (Map.Entry<String, Attribute> entry : legacy.entrySet()) {
+			String legacyKey = entry.getKey();
+			Attribute attribute = entry.getValue();
+
+			attributeNameMap.put(legacyKey.replaceFirst(".", "_"), attribute);
+			attributeNameMap.put(legacyKey.replaceAll("[._]", ""), attribute);
+		}
```

Comment:
Note to contributors: PR will be merge-ready when this is tested with the first builds of PaperMC.